src/SupabaseClient.ts

```diff
@@ -90,19 +98,29 @@ export default class SupabaseClient<
     }
     // default storage key uses the supabase project ref as a namespace
     const defaultStorageKey = `sb-${new URL(this.authUrl).hostname.split('.')[0]}-auth-token`
-    this.storageKey = options?.auth?.storageKey ?? defaultStorageKey
+    const DEFAULTS = {
+      db: DEFAULT_DB_OPTIONS,
+      realtime: DEFAULT_REALTIME_OPTIONS,
+      auth: { ...DEFAULT_AUTH_OPTIONS, storageKey: defaultStorageKey },
+      global: DEFAULT_GLOBAL_OPTIONS,
+    }
 
-    const settings = { ...DEFAULT_OPTIONS, ...options, storageKey: this.storageKey }
+    const settings = applySettingDefaults(options || {}, DEFAULTS)
 
-    this.headers = { ...DEFAULT_HEADERS, ...options?.headers }
+    this.storageKey = settings.auth?.storageKey ?? ''
```

Comment:
will storageKey ever be nullish here since we are setting it to the defaultStorageKey above?

Reply:
Nope,  you're right that it shouldn't be Nullish. However, I think the compiler thinks it might be Null-ish because `applySettingDefaults` returns `SupabaseClientOptions` which allows for null-ish storage key.

This is probably something that should be fixed in the typing (e.g. via a new type or other), will leave it there and merge for now(so as not to block other changes) but am open to suggestions on how we can best fix.